src/main/java/bio/terra/service/filedata/azure/tables/TableServiceClientUtils.java

```diff
@@ -0,0 +1,35 @@
+package bio.terra.service.filedata.azure.tables;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+
+public class TableServiceClientUtils {
+
+  public static boolean tableHasEntries(
+      TableServiceClient tableServiceClient, String tableName, ListEntitiesOptions options) {
+    if (tableExists(tableServiceClient, tableName)) {
+      TableClient tableClient = tableServiceClient.getTableClient(tableName);
+
+      PagedIterable<TableEntity> tableEntities = tableClient.listEntities(options, null, null);
+      if (tableEntities.iterator().hasNext()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public static boolean tableHasEntries(TableServiceClient tableServiceClient, String tableName) {
+    ListEntitiesOptions options = new ListEntitiesOptions();
+    return tableHasEntries(tableServiceClient, tableName, options);
+  }
+
+  public static boolean tableExists(TableServiceClient tableServiceClient, String tableName) {
+    return tableServiceClient.listTables().stream()
+        .filter(table -> table.getName().matches(tableName))
```

Comment:
I think this may not be the optimal way of do it. You are returning all tables and then applying the filter.  The preference would be to have the service do the filter. Some along these lines:

```java
ListTablesOptions options = new ListTablesOptions().setFilter("TableName eq 'myTable'");
 
 PagedIterable<TableItem> retrievedTableItems = tableServiceClient.listTables(options, Duration.ofSeconds(5),
     new Context("key1", "value1"));
 
 ```